exp/controllers/awsmachinepool_machines.go

```diff
@@ -161,7 +160,7 @@ func deleteOrphanedAWSMachines(ctx context.Context, awsMachineList *infrav1.AWSM
 	return nil
 }
 
-func getAWSMachines(ctx context.Context, mp *expclusterv1.MachinePool, kubeClient client.Client) (*infrav1.AWSMachineList, error) {
+func getAWSMachines(ctx context.Context, mp *clusterv1.MachinePool, kubeClient client.Client) (*infrav1.AWSMachineList, error) {
 	if !feature.Gates.Enabled(feature.MachinePoolMachines) {
 		return nil, errors.New("getAWSMachines must not be called unless the MachinePoolMachines feature gate is enabled")
 	}
```

Comment:
nit: also here we are listing AWSMachine's and thus should use clusterv1beta1?

Reply:
Not sure about this one, changing to `v1beta1.MachinePool` here will force to change `scope.MachinePoolScope` as well (or do conversion).

Also this is not changing any API, I tried to use the new types wherever possible as long as it does not change the public API.

What is the downside here?

Reply:
I will leave it with the new API then unless there is objection from the maintainers.